### PR TITLE
Update project dependencies

### PR DESCRIPTION
**Pull request by friederbluemle**
This updates all Android build dependencies to the latest versions (as of March 2nd 2017).

<sub><sup>Please NO GitHub rebase/squash merges. If you'd like me to combine some or all of the commits, please let me know, and I will update my branch. Thank you.</sup></sub>

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -9,8 +9,7 @@
 
 # Specifies the JVM arguments used for the daemon process.
 # The setting is particularly useful for tweaking memory settings.
-# Default value: -Xmx10248m -XX:MaxPermSize=256m
-# org.gradle.jvmargs=-Xmx2048m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
+org.gradle.jvmargs=-Xmx1536m
```
Comment (stkent) [inline gradle.properties:12]: What was the reasoning behind this change?

File: example/build.gradle [hunk 2]
```diff
@@ -30,11 +30,6 @@ android {
     }
 
     buildTypes {
-        debug {
-            minifyEnabled true
-            proguardFiles getDefaultProguardFile('proguard-android.txt')
-        }
```
Comment (stkent) [inline example/build.gradle]: Why was this removed? This library relies on reflection (and there's an open issue to introduce a variant of the example app that relies on said reflection), so I'd like this to remain.

**Review by stkent — CHANGES_REQUESTED** [review]
Thanks for the contribution! I've commented in a couple spots where I don't understand the reasoning behind the changes. Everything else looks good!

**Comment by friederbluemle** [comment]
Sure. Let me add some explanations. :)

File: gradle.properties (hunk 1 above)
Comment (friederbluemle) [reply gradle.properties:12]: The reason is that with the default setting of 1GB and a more recent version of the Android build tool chain and Gradle, `dex` will run out of process since it requires a larger heap. Adding this setting allows dex to run in the same process, speeding up the build. This is now also the default setting, if you create a new project using the latest version of Android Studio.

During the build, you would encounter this warning if the heap was not explicity set to at least 1.5G:

```
Running dex as a separate process.

To run dex in process, the Gradle daemon needs a larger heap.
It currently has 1024 MB.
For faster builds, increase the maximum heap size for the Gradle daemon to at least 1536 MB.
To do this set org.gradle.jvmargs=-Xmx1536M in the project gradle.properties.
For more information see https://docs.gradle.org/current/userguide/build_environment.html
```

File: example/build.gradle (hunk 2 above)
Comment (friederbluemle) [reply example/build.gradle]: Good question and sorry for removing without an explanation. When I saw it I assumed it was there by accident, since it's extremely unusual to run ProGuard on the *debug* variant of a *sample* project. Why exactly is it needed?

**Comment by friederbluemle** [comment]
The stable version of Android Studio 2.3.0 and the matching Gradle plugin were released today - I amended `Update Gradle wrapper and plugin` using a fixup commit, to reflect the latest version.

File: gradle.properties (hunk 1 above)
Comment (stkent) [reply gradle.properties:12]: Thanks, this is a really great and thorough explanation!

File: example/build.gradle (hunk 2 above)
Comment (stkent) [reply example/build.gradle]: It's mostly a sanity check for myself while working with debug builds, checking that the consumer ProGuard files are doing their job appropriately. In general I like to match ProGuard configurations between debug and release builds anyway, to minimize fun runtime release-only surprises.

BugShaker uses reflection internally to decide whether or not the embedding application depends on Google Play Services. This avoids needing to compile Play Services Maps classes into BugShaker. If ProGuard was not configured properly, the class names used to make this determination would be obscured and the library would fail to (attempt to) capture maps in screenshots. See https://github.com/stkent/bugshaker-android/blob/b93fe8455a41b2f696d48794832754a4d48a6f44/bugshaker/src/main/java/com/github/stkent/bugshaker/BugShaker.java#L281-L284 for the actual check. There may well be a better/more robust way to perform this check, in which case, I'm all ears!

**Comment by stkent** [comment]
> The stable version of Android Studio 2.3.0 and the matching Gradle plugin were released today - I amended `Update Gradle wrapper and plugin` using a fixup commit, to reflect the latest version.

Thanks!

File: example/build.gradle (hunk 2 above)
Comment (friederbluemle) [reply example/build.gradle]: Okay thanks for the detailed explanation! I've removed the commit from my branch.

**Comment by stkent** [comment]
Thanks @friederbluemle :D
